source/utils/ethUtil.ts

```diff
@@ -1,106 +1,66 @@
-import { useController } from 'hooks/index';
 import InputDataDecoder from 'ethereum-input-data-decoder';
-import Web3 from 'web3';
-import { KeyringNetwork } from '@stardust-collective/dag4-keyring'
 import store from 'state/store';
 import IVaultState, { AssetType, IAssetState } from 'state/vault/types';
-import erc20ABI from './erc20.json'
-import { getContractDetails } from './etherscan';
-import { EthNetworkId } from 'scripts/Background/controllers/EthChainController/types';
-import { getInfuraUrl } from 'scripts/Background/controllers/EthChainController/utils';
+import erc20ABI from './erc20.json';
+import { getWalletController } from './controllersUtils';
+import { parseUnits } from 'ethers/lib/utils';
 
 export const getERC20DataDecoder = () => {
     return new InputDataDecoder(erc20ABI as any);
 }
 
 export const estimateGasPrice = async (): Promise<number> => {
-    const controller = useController();
+    const walletController = getWalletController();
+    const ethClient = walletController.account.ethClient;
 
-    const client = controller.wallet.account.ethClient;
-
-    if (!client) {
+    if (!ethClient) {
         return 0;
     }
 
     // Returns average, fast, or fastest
-    const { fast } = await client.estimateGasPrices();
+    const { fast } = await ethClient.estimateGasPrices();
     return fast.amount().toNumber();
 }
```

Comment:
This function is not being used at all, follow [`EthereumProvider.getGasEstimate()`](https://github.com/StardustCollective/stargazer-wallet-ext/blob/ab31e010f52915b94b6f8cad39e1557df0d8bc06/source/scripts/Provider/EthereumProvider.ts#L101). We should remove both.